client/my-sites/plugins/controller-logged-out.js

```diff
@@ -36,11 +41,49 @@ const prefetchPopularPlugins = ( queryClient, options ) =>
 const prefetchFeaturedPlugins = ( queryClient ) =>
 	prefetchPluginsData( queryClient, getWPCOMFeaturedPluginsQueryParams() );
 
-const prefetchProductList = ( store ) => {
-	const productsList = getProductsList( store.getState() );
-	if ( Object.values( productsList ).length === 0 ) {
-		return requestProductsList( { type: 'all' } )( store.dispatch );
-	}
+const prefetchProductList = ( queryClient, store ) => {
+	const query = { type: 'all' };
+
+	return queryClient
+		.fetchQuery( 'products-list', () => wpcom.req.get( '/products', query ) )
```

Comment:
The `productsList` data prefetch has been wrapped inside a `react-query` cache so that:
* it can get cached if a timeout happened
* if allows the same instance to be used on all pages (cacheKeys)

Reply:
The key should be `[ 'products-list', query.type ]`, in case query with another type is issued.